### ISSUE TITLE
Restart job at workflow level

### ISSUE THREAD

**Issue by huikang** [argoproj/argo#1578]
I have an argo workflow that is triggered through argo-event, If anyone of the steps in the workflow failed, how could the workflow be retried automatically. Is there any way to setup a retry strategy at the workflow level? Thanks.

https://argoproj.slack.com/archives/C8J6SGN12/p1566439599263600

**Comment by dtaniwaki** [comment]
I think this feature is very useful for me too. May I implement it?

**Comment by dtaniwaki** [comment]
How about allowing `retryStrategy` in templates other than container templates?

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: workflow-template-retry-workflow-
spec:
  entrypoint: retry-workflow
  templates:
  - name: random-fail-template
    container:
      image: python:alpine3.6
      command: [python, -c]
      # fail with a 66% probability
      args: ["import random; import sys; exit_code = random.choice([0, 1, 1]); sys.exit(exit_code)"]
  - name: retry-workflow
    template: steps
    retryStrategy:
      limit: 10
  - name: steps
    steps:
    - - name: hello1
        template: random-fail-template
    - - name: hello2a
        template: random-fail-template
      - name: hello2b
        template: random-fail-template
```

**Comment by epa095** [comment]
@dtaniwaki that would be good yes, there is really no reason not to allow retrying on other  template-types than container. I previously made an issue for retrying resource-templates (#1370), but it holds for all template types really.

**Comment by nicorikken** [comment]
Restarting workflows is a feature in v2.5.0.

**Comment by simster7** [comment]
Closing

**Comment by huikang** [comment]
@nicorikken , @simster7 , where can I find the example to retry the whole workflow pipeline if one step fails? I searched, but only find examples about retrying one step. Thanks.

**Comment by simster7** [comment]
This issue was closed on error, the feature has not been implemented

**Comment by alexec** [comment]
Fixed in v2.12
